src/VS.Mefx/Commands/Command.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace VS.Mefx.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.Composition;
+
+    /// <summary>
+    /// A general command class which serves a parent class for all the commands that can be run by application .
+    /// </summary>
+    internal class Command
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Command"/> class.
+        /// </summary>
+        /// <param name="derivedInfo">The ConfigCreator for the input files.</param>
+        /// <param name="arguments">The command line arguments from the user.</param>
+        public Command(ConfigCreator derivedInfo, CLIOptions arguments)
+        {
+            this.Creator = derivedInfo;
+            this.Options = arguments;
+        }
+
+        /// <summary>
+        /// Write the given lines in sorted order to the user.
+        /// </summary>
+        /// <param name="lines">The lines to output to the user.</param>
+        protected void WriteLines(List<string> lines)
+        {
+            lines.Sort();
+            lines.ForEach(line => this.Options.Writer.WriteLine(line));
+        }
+
+        /// <summary>
+        /// Gets the composable catalog and configuration.
+        /// </summary>
+        protected ConfigCreator Creator { get; private set; } // Stores the catalog and config information for the input files.
```

Comment:
I think you can drop the `private set;` since only the ctor sets this property.
Also, why do you have the `//` code comment as well as the `<summary>` comment? Is the summary not correct or sufficient?